File: pythonPractice/NaiveBayes.py
def compute_probabilities(data):
    pass_count = sum(1 for row in data if row[-1] == "yes")
    fail_count = len(data) - pass_count
    p_pass = pass_count / len(data)
    p_fail = fail_count / len(data)
    return p_pass, p_fail

def test_classifier(data, test):
    p_pass, p_fail = compute_probabilities(data)
    for row in data:
        if row[:-1] == test:
            if row[-1] == "yes":
                p_pass *= pass_count / len(data)
            else:
                p_fail *= fail_count / len(data)
    p_pass /= (p_pass + p_fail)
    p_fail /= (p_pass + p_fail)
    prediction = "yes" if p_pass > p_fail else "no"
    return prediction, p_pass, p_fail

data = [
    ["<=30","high","no","fair","no"],
    ["<=30","high","no","excellent","no"],
    ["31...40","high","no","fair","yes"],
    [">40","medium","no","fair","yes"],
    [">40","low","yes","fair","yes"],
    [">40","low","yes","excellent","no"],
    ["31...40","low","yes","excellent","yes"],
    ["<=30","medium","no","fair","no"],
    ["<=30","low","yes","fair","yes"],
    [">40","medium","yes","fair","yes"],
    ["<=30","medium","yes","excellent","yes"],
    ["31...40","medium","no","excellent","yes"],
    ["31...40","high","yes","fair","yes"],
    [">40","medium","no","excellent","no"]
]

test = ["<=30", "high", "no", "fair"]  # Replace with your test data
prediction, p_pass, p_fail = test_classifier(data, test)

print("Test Data:", test)
print("Prediction:", prediction)
print("Probability of 'yes':", p_pass)
print("Probability of 'no':", p_fail)

File: practice Codes/NaiveBayes.java
import java.util.Scanner;

public class NaiveBayes {
    static String[][] data = {
            { "Good", "High", "A", "Pass" },
            { "Good", "Low", "B", "Pass" },
            { "Poor", "High", "C", "Fail" },
            { "Poor", "Low", "B", "Fail" },
            { "Good", "Low", "A", "Pass" },
            { "Good", "High", "C", "Pass" }
    };
    static double P_PASS;
    static double P_FAIL;

    public static void main(String[] args) {
        computeProbabilities();
        testClassifier();
    }

    public static void computeProbabilities() {
        int passCount = 0;
        int failCount = 0;
        int totalStudents = data.length;
        for (String[] row : data) {
            String study = row[0];
            String sports = row[1];
            String grade = row[2];
            String result = row[3];
            if (result.equals("Pass")) {
                passCount++;
            } else {
                failCount++;
            }
        }
        P_PASS = (double) passCount / totalStudents;
        P_FAIL = (double) failCount / totalStudents;
    }

    public static void testClassifier() {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter comma-separated test tuple (e.g., 'Good,High,A'): ");
        String input = scanner.nextLine();
        String[] test = input.split(",");
        double pPass = P_PASS;
        double pFail = P_FAIL;
        for (String[] row : data) {
            String study = row[0];
            String sports = row[1];
            String grade = row[2];
            String result = row[3];
            if (study.equals(test[0]) && sports.equals(test[1]) && grade.equals(test[2])) {
                if (result.equals("Pass")) {
                    pPass++;
                } else {
                    pFail++;
                }
            }
        }
        pPass /= (P_PASS + P_FAIL);
        pFail /= (P_PASS + P_FAIL);
        System.out.println("Probability of Passing: " + pPass);
        System.out.println("Probability of Failing: " + pFail);
        if (pPass > pFail) {
            System.out.println("Prediction: Pass");
        } else {
            System.out.println("Prediction: Fail");
        }
    }
}